include/device.h

```diff
@@ -781,15 +762,12 @@ static inline int device_set_power_state(const struct device *dev,
 					 uint32_t device_power_state,
 					 device_pm_cb cb, void *arg)
 {
-	if (dev->device_pm_control) {
-		return dev->device_pm_control(dev,
-						 DEVICE_PM_SET_POWER_STATE,
-						 &device_power_state, cb, arg);
-	} else {
-		return device_pm_control_nop(dev,
-						 DEVICE_PM_SET_POWER_STATE,
-						 &device_power_state, cb, arg);
+	if (dev->device_pm_control ==  NULL) {
+		return -ENOTSUP;
```

Comment:
should return -ENOSYS (not implemented)

Reply:
fixed, thanks (note: `-ENOTSUP` was returned from `device_pm_control_nop`).